### PR TITLE
Fix: make analyze-commits logic platform agnostic

### PR DESCRIPTION
**Pull request by alisd23**
This PR is to fix the commit analyzer for Windows (and hopefully other platforms).
There was some logic which assumed Unix paths which I have modified.

- Remove any hard-coded paths/path segments.
- Use the `path` library for any path manipulation, including normalizing paths for the host OS.
- Also fixed a bug where if a package name is a substring of another package, commits could be filtered incorrectly. E.g. `my-package` and `my-package-2`.

### CODE REVIEW

File: src/log-plugin-version.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
-const { resolve } = require('path');
+const path = require('path');
 const readPkg = require('read-pkg');
 const debug = require('debug')('semantic-release:monorepo');
 
 const logPluginVersion = type => plugin => async (pluginConfig, config) => {
   if (config.options.debug) {
-    const { version } = await readPkg(resolve(__dirname, '../'));
+    const { version } = await readPkg(path.resolve(__dirname, '../'));
```
Comment (wtgtybhertgeghgtwtg) [inline src/log-plugin-version.js]: What does this have to do with this PR?

Comment (alisd23) [reply src/log-plugin-version.js]: Not a lot, just thought I'd make the path imports consistent - but actually maybe I did it the wrong way round. I'll revert it.

**Comment by pmowrer** [comment]
Thanks for this, @alisd23  and @wtgtybhertgeghgtwtg ! I just want to verify that it works on our end. I'll try to have this merged by EOW!
